src/main/java/org/truffleruby/core/exception/SyntaxErrorNodes.java

```diff
@@ -35,4 +40,26 @@ RubySyntaxError allocateSyntaxError(RubyClass rubyClass) {
 
     }
 
+    @CoreMethod(names = "path")
+    public abstract static class PathNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
```

Comment:
This should be a `@TruffleBoundary` annotation on top, because several boundary methods are called below so it's more efficient this way.

Reply:
Thank you! By boundary method do you mean those imported from `com.oracle.truffle.api`? Where can I learn more about this?

Reply:
By boundary methods I mean methods annotated with `@TruffleBoundary`